alacritty/src/window_context.rs

```diff
@@ -167,33 +182,49 @@ impl WindowContext {
     }
 
     /// Update the terminal window to the latest config.
-    pub fn update_config(&mut self, old_config: &UiConfig, config: &UiConfig) {
-        self.display.update_config(config);
-        self.terminal.lock().update_config(&config.terminal_config);
+    pub fn update_config(&mut self, new_config: Rc<UiConfig>) {
+        let old_config = mem::replace(&mut self.config, new_config);
+
+        // Apply ipc config if there are any overrides.
```

Comment:
```suggestion
        // Apply ipc config if there are overrides.
```

It kind of reads redundant, but I don't have a strong opinion on this one.